src/search.c

```diff
@@ -805,10 +805,10 @@ static int download_manifests(struct manifest **MoM, struct list **subs)
 		return ECURRENT_VERSION;
 	}
 
-	*MoM = load_mom(current_version, false, false);
+	*MoM = load_mom(current_version, false, false, NULL);
```

Comment:
My understanding of #323 is that part of the issue is error code propagation. Is there a reason that we still want to not get the exact error codes from load_mom? We could return the error code from load_mom instead of using EMOM_LOAD as a generic error.

Reply:
The other day I had a talk with @otaviobp about this, we talked about what return codes we should be returning back to the end user. He mentioned that although we can use many different codes within the functions of the swupd code, the codes we are supposed to return to the end user are only those listed in https://github.com/clearlinux/swupd-client/blob/master/src/swupd-error.h.
This make sense to me, because at the moment internally we also use error codes from curl or from errno.h or errno-base.h, so if we propagate those errors to the end user they may not make any sense to the user anyway. 

I understand that part of #323 is error code propagation, and that part has been addressed by adding the exact error code to the load_mom function.  But in the end we need to translate that error into one of the swupd-errors before returning to the user, so we cannot really always propagate all the codes up to the user.

I think the most important part of what #323 complains about is the fact that the error returned was misleading and the presented output didn't provide much help to determine what caused the error. I think that has been fixed now with the changes in the PR, even if the return code received by the end user is a generic EMOM_LOAD error (which is more accureate than EMOM_NOTFOUND) the messages that are printed as part of the output will now be useful to determine the problem.

Reply:
I understand and this looks good to me now. Thanks for the clarification.

Reply:
I would add just one thing to @castulo comment. All error codes returned by a Linux application should be a positive integer number from 0 to 255, where 0 is success. And all exit codes should be in swupd-error.h and documented on doc/swupd.1.